app/models/queen.rb

```diff
@@ -1,2 +1,20 @@
 class Queen < Piece
+  def path_to_king
+    king = game.pieces.where.not(is_black: is_black).where(type: 'King').first
+    return nil unless king.row == row || king.col == col ||
+                      (king.row - row).abs == (king.col - col).abs
+    get_path(king.row, king.col)
+  end
+
+  def get_path(to_row, to_col)
```

Comment:
I think the `get_path` method is the same in the Queen and Bishop - it can probably be moved to the Piece class even if it's not used by every piece.